items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -22,177 +23,190 @@ import scala.concurrent.{ExecutionContext, Future}
-  ): DisplayItem = {
-    val updatedItemLocations = item.locations.map {
-      case physicalLocation: DisplayPhysicalLocation =>
-        physicalLocation.copy(accessConditions = List(accessCondition))
-      case location => location
+    accessCondition: Option[DisplayAccessCondition]
+  ): DisplayItem =
+    accessCondition match {
+      case Some(accessCondition) =>
+        val updatedItemLocations = item.locations.map {
+          case physicalLocation: DisplayPhysicalLocation =>
+            physicalLocation.copy(accessConditions = List(accessCondition))
+          case location => location
+        }
+        item.copy(locations = updatedItemLocations)
+      case None => item
     }
 
-    item.copy(locations = updatedItemLocations)
-  }
-
-  private def getAccessConditions(
-    existingItems: Map[SierraItemNumber, Option[DisplayLocationType]]
-  ): Future[Map[SierraItemNumber, DisplayAccessCondition]] =
-    for {
-      itemEither <- sierraSource.lookupItemEntries(existingItems.keys.toSeq)
-
-      maybeAccessConditions: Map[
-        SierraItemNumber,
-        Option[
-          DisplayAccessCondition
-        ]] = itemEither match {
-        case Right(SierraItemDataEntries(_, _, entries)) =>
-          entries
-            .map(item => {
-              val location = existingItems.get(item.id).flatten
-              item.id -> item.accessCondition(location)
-            })
-            .toMap
-        case Left(
-            SierraItemLookupError.MissingItems(missingItems, itemsReturned)
-            ) =>
-          warn(s"Item lookup missing items: $missingItems")
-          itemsReturned
-            .map(item => {
-              val location = existingItems.get(item.id).flatten
-              item.id -> item.accessCondition(location)
-            })
-            .toMap
-        case Left(itemLookupError) =>
-          error(s"Item lookup failed: $itemLookupError")
-          Map.empty[SierraItemNumber, Option[DisplayAccessCondition]]
-      }
-
-      accessConditions = maybeAccessConditions
-        .collect { case (itemId, Some(ac)) => itemId -> ac }
-
-    } yield accessConditions
-
+  /** Set availability slots for a single item
+    *  - if its physicalAccessCondition exists and is requestable
+    *  - based on its location
+    *  If any of the above are not true/defined, we return the item without availableDates
+    */
   private def setAvailableDates(
-    item: DisplayItem
-  ): Future[DisplayItem] = {
-    // there is only ever one location per physicalItem and one accessCondition per location,
-    // but this may sometimes be empty if it could not be fetched.
-    val accessCondition = item.locations.headOption
-      .flatMap(_.accessConditions.headOption)
-
-    if (accessCondition.exists(_.isRequestable)) {
-      getAvailableDates(accessCondition.head)
-        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+    item: DisplayItem,
+    sierraItemLocation: Option[SierraLocation]
```

Comment:
I think you might be able to simplify by checking for a `Some` and unwrapping the option in the calling function, not blocking for merge though.

Reply:
I pushed it down because the calling function `updateItem` ended up with multiple nested match cases and I thought it looked a bit tangled